core/cap-0067.md

```diff
@@ -136,6 +136,50 @@ index 0fc03e2..963acc4 100644
  // This struct groups together changes on a per transaction basis
 ```
 
+### New host functions
+
+The diff is based on commit `9a24835ec6d75c526fb128a1d73b92e7d7becfa7` of `rs-soroban-env`.
+
+```diff mddiffcheck.ignore=true
+diff --git a/soroban-env-common/env.json b/soroban-env-common/env.json
+index d421dca2..4249efc6 100644
+--- a/soroban-env-common/env.json
++++ b/soroban-env-common/env.json
+@@ -2403,6 +2403,32 @@
+                     ],
+                     "return": "Void",
+                     "docs": "Authorizes sub-contract calls for the next contract call on behalf of the current contract. Every entry in the argument vector corresponds to `InvokerContractAuthEntry` contract type that authorizes a tree of `require_auth` calls on behalf of the current contract. The entries must not contain any authorizations for the direct contract call, i.e. if current contract needs to call contract function F1 that calls function F2 both of which require auth, only F2 should be present in `auth_entries`."
++                },
++                {
++                    "export": "4",
++                    "name": "get_address_from_muxed_address",
```

Comment:
Would someone ever need one but not the other? `split_muxed_address` might be more ergonomic. No strong opinion, just a suggestion - if there's a use case for one xor the other, this is moot.

Reply:
That was my original intuition as well, but since we'd need to return the parts as a host vector, we'd need to do a bunch of additional host calls that aren't really that ergonomic. So this is actually more ergonomic, at least from the host API standpoint.

Reply:
Sure, makes sense, ty for explanation :+1: